packages/machine/src/models/app-instance.ts

```diff
@@ -0,0 +1,231 @@
+import { AppIdentity, AppInterface, Terms } from "@counterfactual/types";
+
+import { defaultAbiCoder, keccak256, solidityPack } from "ethers/utils";
+import { Memoize } from "typescript-memoize";
+
+import { appIdentityToHash } from "../ethereum/utils/app-identity";
+import { APP_INTERFACE, TERMS } from "../ethereum/utils/encodings";
+
+/**
+ * Represenation of the values a dependency nonce can take on.
+ */
+export enum DependencyValue {
+  NOT_UNINSTALLED = 0,
```

Comment:
I did it this way because the _default_ status of the value is that an app has not explicitly been uninstalled yet. There is no step that changes this value from “nothing” to “installed”. There is only a step from “not uninstalled” to “uninsyalled”